### DISCUSSION TITLE
Does WinDynamicDesktop need to be constantly running for the wallpapers to dynamically cycle?

### DISCUSSION THREAD

**Discussion by FoxRunTime** [t1m0thyj/WinDynamicDesktop#282] [Q&A] [answered]
Ironically, I've recently switched from macOS to Windows 10, but there are a few things I miss from macOS, the Dynamic Wallpapers being one of them.
I found this, and I'm super happy that it exists! But does the app have constantly be running in the background for the wallpapers to dynamically cycle, or can I set a theme and exit the app?

Thanks!

**Comment by t1m0thyj** [comment, marked as answer]
> But does the app have constantly be running in the background for the wallpapers to dynamically cycle, or can I set a theme and exit the app?

Yes it does. Windows doesn't have native support for changing wallpapers based on the time of day, so WDD must be running in the background to do so. It should be light enough on resources that you can forget about it being there. And if you really want to forget about it being there, you can also [hide the taskbar icon](https://github.com/t1m0thyj/WinDynamicDesktop/wiki/Frequently-asked-questions#how-can-i-hide-the-tray-icon) 🙂

**Reply by ze0ss** [reply]
Any significant battery life impact ? And cpu, ram usage?

**Reply by t1m0thyj** [reply]
> Any significant battery life impact ? And cpu, ram usage?

Impact on all of these should be insignificant. You can see the CPU and RAM usage for yourself in Task Manager.

The app sits dormant most of the time. It only uses resources momentarily when a timer fires and it updates your desktop wallpaper.
